generic/piranha/test-resources/kt/feature_flag_system_1/control/input/XPFlagCleanerCases.kt

```diff
@@ -0,0 +1,224 @@
+ limitations under the License.
+*/
+
+package com.uber.input
+
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+
+internal class XPFlagCleanerPositiveCases {
+    enum class TestExperimentName {
+        
+        STALE_FLAG
+    }
+
+    enum class AnotherTestExperimentName {
+        
+        @Autorollout
+        STALE_FLAG
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    annotation class Autorollout(val staged: Boolean = false)
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
+    internal annotation class ToggleTesting(val treated: Array<TestExperimentName>)
+
+    private val experimentation: XPTest? = null
+    private var tBool = false
+    fun conditional_contains_stale_flag() {
+        
+        if (experimentation!!.isToggleEnabled(TestExperimentName.STALE_FLAG)) {
+            println("Hello World")
+        }
+    }
+
+    fun conditional_with_else_contains_stale_flag() {
+        if (experimentation!!.isToggleEnabled(TestExperimentName.STALE_FLAG)) {
+            println("Hello World")
+        } else {
+            println("Hi world")
+        }
+    }
+
+     fun complex_conditional_contains_stale_flag() {
+        if (true || tBool && experimentation!!.isToggleEnabled(TestExperimentName.STALE_FLAG)) {
+            println("Hello World")
+        } else {
+            println("Hi world")
+        }
+    }
+
+    fun other_api_stale_flag() {
+        if (experimentation!!.isFlagTreated(TestExperimentName.STALE_FLAG)) {
+            println("Hello World")
+        } else {
+            println("Hi world")
+        }
+    }
+
+    fun assignments_containing_stale_flag() {
+        
+        tBool = experimentation!!.isToggleEnabled(TestExperimentName.STALE_FLAG)
+
+        
+        tBool = experimentation.isToggleEnabled(TestExperimentName.STALE_FLAG) && true
+
+        
+        tBool = experimentation.isToggleEnabled(TestExperimentName.STALE_FLAG) || true
+
+        
+        tBool = experimentation.isToggleEnabled(TestExperimentName.STALE_FLAG) || tBool
+
+        
+        tBool = experimentation.isToggleEnabled(TestExperimentName.STALE_FLAG) && (tBool || true)
+    }
+
+    fun return_contains_stale_flag(): Boolean {
+        // FIXME:
+        return experimentation!!.isToggleEnabled(TestExperimentName.STALE_FLAG)
+    }
+
+    fun condexp_contains_stale_flag() {
+        
+        tBool = if (experimentation!!.isToggleEnabled(TestExperimentName.STALE_FLAG)) true else false
+    }
+
+    fun misc_xp_apis_containing_stale_flag() {
+        
+        if (experimentation!!.isToggleEnabled(TestExperimentName.STALE_FLAG) && (tBool || true)) {
+        }
+
+        
+        experimentation.putToggleEnabled(TestExperimentName.STALE_FLAG)
+
+        
+        experimentation.includeEvent(TestExperimentName.STALE_FLAG)
+
+        
```

Comment:
it will great if this file can be cleaned up by deleting unnecessary empty lines.